config/config.go

```diff
@@ -155,6 +157,9 @@ type Service struct {
 
 	// Pipelines are the set of data pipelines configured for the service.
 	Pipelines Pipelines
+
+	// Gates is an inferface that can be queried regarding the state of feature gates.
+	Gates featuregate.Gates
```

Comment:
There is something that I don't like about config depending on the "service/featuregate". In the config we tried to keep only "raw configuration", so probably based on how the config looks like this should be "map[string]bool". That is the "Configuration" for the gates.

Reply:
I kind of feel that this is "the result" of running the service.